### PR TITLE
Fetch account balance from RPC node

### PR DESCRIPTION
**Pull request by lubej**
Account that this can be tested on: `<explorer-url>/testnet/sapphire/address/0x2B95834655499B3C3817f379e1C8E132baa965F4`

### CODE REVIEW

**Comment by lubej** [comment]
Upon reverting this, `ethers` dependency should be removed.

File: src/app/utils/rpc-utils.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    [Layer.emerald]: 'https://emerald.oasis.dev',
+    [Layer.sapphire]: 'https://sapphire.oasis.io',
+  },
+  [Network.testnet]: {
+    [Layer.emerald]: 'https://testnet.emerald.oasis.dev',
+    [Layer.sapphire]: 'https://testnet.sapphire.oasis.dev',
+  },
+}
+
+const NETWORK_DECIMALS: Partial<Record<Layer, number>> = {
```
Comment (lukaw3d) [inline src/app/utils/rpc-utils.ts:17]: LAYER_DECIMALS?

File: src/app/pages/AccountDetailsPage/hook.ts [hunk 2]
```diff
@@ -11,15 +11,66 @@ import { NUMBER_OF_ITEMS_ON_SEPARATE_PAGE } from '../../config'
...
   if (layer === Layer.consensus) {
     // There can be no ERC-20 or ERC-721 tokens on consensus
     throw AppErrors.UnsupportedLayer
   }
   const query = useGetRuntimeAccountsAddress(network, layer, address)
-  const account = query.data?.data
+  const runtimeAccount = query.data?.data
+
+  // TODO: Remove after account balances on Nexus are in sync with the node
+  useEffect(() => {
```
Comment (lukaw3d) [inline src/app/pages/AccountDetailsPage/hook.ts:29]: This works in https://a81a9fac.oasis-explorer.pages.dev/testnet/sapphire/address/0x2B95834655499B3C3817f379e1C8E132baa965F4
but not in https://a81a9fac.oasis-explorer.pages.dev/search?q=0x2B95834655499B3C3817f379e1C8E132baa965F4

Maybe move up into useGetRuntimeAccountsAddress

File: src/app/pages/AccountDetailsPage/hook.ts [hunk 3]
```diff
@@ -11,15 +11,66 @@ import { NUMBER_OF_ITEMS_ON_SEPARATE_PAGE } from '../../config'
...
+
+      const balance = await RpcUtils.getAccountBalance(address, {
+        context: {
+          network: scope.network,
+          layer: scope.layer,
+        },
+      })
+
+      if (shouldUpdate) {
+        setAccountBalance(balance)
```
Comment (lukaw3d) [inline src/app/pages/AccountDetailsPage/hook.ts:48]: (wrapping it into react-query would have been prettier, but this is temporary)

File: src/app/utils/rpc-utils.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  static async getAccountBalance(address: string, opts: RpcProviderOptions): Promise<string> {
+    const provider = RpcUtils._getProvider(opts)
+
+    const balance = await provider.getBalance(address)
+
+    const {
+      context: { layer },
+    } = opts
+    const decimals = NETWORK_DECIMALS[layer] ?? 18
```
Comment (lukaw3d) [inline src/app/utils/rpc-utils.ts:64]: I would remove the `?? 18` fallback and just add `consensus: 9` to layer decimals

**Review by lukaw3d — APPROVED** [review]

**Pull request merged**